core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -743,6 +743,18 @@ abstract class AbstractFetcherThread(name: String,
     } finally partitionMapLock.unlock()
   }
 
+  def removeAllPartitions(): Map[TopicPartition, PartitionFetchState] = {
+    partitionMapLock.lockInterruptibly()
```

Comment:
Maybe we can just feed all partitions into `removePartitions` method?

Reply:
It might release the lock when trying to get all partitions  and use the result to remove might lead to inconsistency(when this fetcher add some partition right between the two process called by `AbstractFetcherManager`. So I make the process into one new method.
Another point is that we don't need to do some filtering or get the state of thread one by one, but just iterate and return the copy of the whole `partitionStates.partitionStateMap`. I think it seems more efficient in realization.

Reply:
Make sense!